lib/facility.rb

```diff
@@ -1,14 +1,113 @@
+    if @services.include?('Vehicle Registration')
+    data.registration_date = Date.today
+    @registered_vehicles.push(data)
+      if data.antique? 
+        data.plate_type = :antique
+        @collected_fees += 25
+      elsif
+        data.electric_vehicle? 
+        data.plate_type = :ev
+        @collected_fees += 200
+      else
+        data.plate_type = :regular
+        @collected_fees += 100
+      end
+      @registered_vehicles
+    end
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def administer_written_test(data)
+    if @services.include?('Written Test')
+      if data.permit? && data.age >= 16
+          data.license_data[:written] = true
+      end
+    end
+      data.license_data[:written]
+  end
+
+  def administer_road_test(data)
+    if @services.include?('Road Test')
+      if data.license_data[:written] == true
+          data.license_data[:license] = true
+      end
+    end
+      data.license_data[:license]
+  end
+
+  def renew_drivers_license(data)
+    if @services.include?('Renew License')
+      if data.license_data[:written] == true 
+          data.license_data[:license] = true
+            data.license_data[:renewed] = true
+      end
+    end
+      data.license_data[:renewed]
+  end
+
+  def parse_name(office_details)
+    if office_details[:name]
+        office_details[:name]
+    elsif office_details[:title] 
+          office_details[:title] 
+    elsif office_details[:office_name]
+          office_details[:office_name]
+    end
+  end
+
+  def parse_address(office_details)
+    if office_details[:address]
+        office_details[:address]
+    elsif office_details[:location_1]
+          address = Hash.new
+          location_1 = JSON.parse office_details[:location_1][:human_address]
+          address.store(:new_address, location_1)
+          if office_details[:location_2]
+            address_2 = Hash.new
+                location_2 = JSON.parse office_details[:location_2][:human_address] 
+                address_2.store(:suite, location_2['address'])
+                address[:new_address]["suite"] = address_2[:suite]
+          end
+          address[:new_address].values.join(', ')
+      elsif office_details[:street_address_line_1]
+            address = office_details[:street_address_line_1], office_details[:street_address_line_2], office_details[:city], office_details[:state], office_details[:zip_code]
+            address.join(', ')
+      elsif office_details[:address1]
+            address = office_details[:address1], office_details[:city], office_details[:state], office_details[:zipcode]
+            address.join(', ')
+      end 
+  end
+
+  def parse_phone(office_details)
+    if office_details[:phone]
+      office_details[:phone]
+    elsif office_details[:phone_number] 
+          office_details[:phone_number]
+    elsif office_details[:public_phone_number]
+          office_details[:public_phone_number]
+    end
+  end
```

Comment:
We will talk a lot more about this theme throughout your journey at Turing, but when deciding where functionality should live, it's good to think about WHO or WHAT(class) should be responsible for what logic. As long as you can defend your design choices, lots of different things work. Parsing the facility data so that you can pass it right into the Facility objects that you make is a creative decision that was cool to see!